### PR TITLE
Change `_get_train_sampler` comment to mention `num_iterations > 1`

### LINKED ISSUE #3327: Potential misleading doc string for _get_train_sampler()
### Reproduction

Hi thanks for the amazing work on speeding up training with smart batching design. 

I think there is an error in the docstring of `_get_train_sampler()` describing how batches are splitted within devices and across devices. @qgallouedec 

Based on the specified example, for a `per_device_batch_size=3`, `num_grad_accum=2` and `num_iterations = 2`, each device should receive an `effective_batch_size =  per_device_batch_size * num_grad_accum = 6`, not what docstring has shown to be 3. The default setting of `split_batches=False` used by `accelerator.prepare()` also ensures that each device receives `effective_batch_size=6`

And then `split_tensor_dict()` method within each device/process divides this batch into two chunks of 3. So the correct docstring illustration of the process should be 

        #                                      |         GPU 0         |         GPU 1         |
        #                                      |  Accum 0  |  Accum 1  |  Accum 0  |  Accum 1  |
        #
        #                 global_step   step    <-───>  num_generations=2
        #                                       <-───────> per_device_train_batch_size=3
        #  grad_accum    ▲  ▲  0          0     [0   0   1   1   2   2]  3   3   4   4   5   5    <- Generate for the whole accumulated batch; store the completions; use the first slice to compute the loss
        #     =2         ▼  |  0          1      0   0   1   1   2   2 [ 3   3   4   4   5   5]   <- Take the stored generations and use the second slice to compute the loss
        #                   |
        #                   |  1          2     [0   0   1   1   2   2]  3   3   4   4   5   5    <- Take the stored generations and use the first slice to compute the loss
        #  num_iterations=2 ▼  1          3      0   0   1   1   2   2 [ 3   3   4   4   5   5]   <- Take the stored generations and use the second slice to compute the loss
        #
        #                      2          4     [6   6   7   7   8   8]  9   9  10  10  11  11    <- Generate for the whole accumulated batch; store the completions; use the first slice to compute the loss
        #                      2          5      6   6   7   7   8   8 [ 9   9  10  10  11  11]   <- ...
        #                                          ...

### System Info

NIL

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

### PR DESCRIPTION
**Pull request by anidoesdev**
…avior

# What does this PR do?

The docstring of `_get_train_sampler` in `GRPOTrainer` contained two issues:

1. The intro line said "The first row shows the first sampled batch, the second row shows
   the second sampled batch" — which is inaccurate. For `steps_per_generation=4`, all four
   rows are *slices* of the same generated batch, not different batches. Fixed to: "Each row
   corresponds to a single training step."

2. The only figure illustrated `num_iterations=1` (using `steps_per_generation=4`), where
   every step uses a unique slice with no repetition. It never showed what happens when
   `num_iterations > 1` — the core GRPO feature where the same generation batch is reused
   for multiple gradient update passes before regenerating. Added a second figure showing
   `num_iterations=2` with the default `steps_per_generation=gradient_accumulation_steps=2`,
   where steps 2 and 3 reuse the completions from steps 0 and 1.

Fixes #3327 

## Before submitting

- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request), Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Please add a link to it if that's the case.

## AI writing disclosure

We welcome the use of AI tools to help with contributions. For transparency and to help us improve our review process, please indicate the level of AI involvement in this PR.

- [x] AI-assisted: some parts were suggested or improved by AI, but the PR was written and reviewed by a human.

## Who can review?
@LeonEricsson @qgallouedec 

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag members/contributors who may be interested in your PR.

---

> [!NOTE]
> **Low Risk**
> Comment-only documentation in trainer files; no runtime or training behavior changes.
> 
> **Overview**
> Updates the `_get_train_sampler` comment in **`GRPOTrainer`** and **`RLOOTrainer`** so the ASCII diagram is described correctly.
> 
> The old intro implied each table row was a separate sampled batch. The new text states that **each row is the per-step batch from `_prepare_inputs`**, that rows inside a **`steps_per_generation`** window are **slices of one generated batch**, and that when **`num_iterations > 1`** that whole block is **reused across optimization passes** before the next regeneration.
> 
> No training, sampler, or buffering logic changes—comments only (fixes #3327).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 50313e58d39c9d18e11d0fb74e239204e64a42be. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/trainer/grpo_trainer.py [hunk 1]
```diff
@@ -1020,6 +1019,25 @@ def _get_train_sampler(self, dataset: Dataset | None = None) -> Sampler:
...
+        #  grad_accum    ▲  ▲  0          0     0   0   1   1   2   2   <- Generate for prompts 0 to 5; store completions; use first slice
+        #     =2         ▼  |  0          1     3   3   4   4   5   5   <- Take stored completions; use second slice
+        #                   |
+        #  num_iterations   |  1          2     0   0   1   1   2   2   <- Take stored completions; reuse first slice (2nd iteration)
+        #     =2            ▼  1          3     3   3   4   4   5   5   <- Take stored completions; reuse second slice (2nd iteration)
+        #
+        #                      2          4     6   6   7   7   8   8   <- Generate for prompts 6 to 11; store completions; use first slice
+        #                      2          5     9   9  10  10  11  11   <- Take stored completions; use second slice
+        #                                          ...
+
```
Comment (LeonEricsson) [inline trl/trainer/grpo_trainer.py]:
```suggestion

```

File: trl/trainer/grpo_trainer.py [hunk 2]
```diff
@@ -1004,8 +1004,7 @@ def _get_train_sampler(self, dataset: Dataset | None = None) -> Sampler:
         # 2. repeats the batch multiple times to allow reusing generations across multiple updates. Refer to
         #    _prepare_inputs to see how the generations are stored and reused.
 
-        # In the following figure, the values are the prompt indices. The first row shows the first sampled batch, the
-        # second row shows the second sampled batch, and so on.
+        # In the following figures, the values are the prompt indices. Each row corresponds to a single training step.
```
Comment (LeonEricsson) [inline trl/trainer/grpo_trainer.py]: I agree that the original text is misleading and that this fix is warranted. However I don't think the second ASCII diagram is necessary. The existing comment already state batch reuse, and _prepare_inputs documents the regeneration period. Instead of the ASCII figure, I'd propose a minor clarification as part of this comment

```
In the following figure, the values are the prompt indices. Each row shows the per-step batch
returned by `_prepare_inputs`; rows within a `steps_per_generation` block are slices of the same
generated batch. When `num_iterations > 1`, that block is reused for multiple optimization passes
before regenerating.
```

**Review by anidoesdev — COMMENTED** [review]
Please check this out @LeonEricsson. Made the changes as suggested

**Comment by qgallouedec** [comment]
thanks can you make the same edits in rloo?

**Review by anidoesdev — COMMENTED** [review]
made the changes in rloo too. please check @qgallouedec

**Review by qgallouedec — APPROVED** [review]
